kaal-presentation/src/main/kotlin/cz/eman/kaal/presentation/fragment/KaalToolbarBindingFragment.kt

```diff
@@ -0,0 +1,180 @@
+@file:Suppress("unused")
+
+package cz.eman.kaal.presentation.fragment
+
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import androidx.annotation.CallSuper
+import androidx.annotation.DrawableRes
+import androidx.annotation.IdRes
+import androidx.annotation.LayoutRes
+import androidx.annotation.MenuRes
+import androidx.annotation.StringRes
+import androidx.appcompat.app.ActionBar
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.databinding.ViewDataBinding
+import cz.eman.kaal.presentation.view.hideKeyboard
+
+/**
+ * Base class for managing toolbar in fragment. It registers / unregisters as Activity Action Bar and sets it up.
+ *
+ * The options menu can be specified both via XML and programmatically in code. For the "programmatically" menu it is
+ * necessary to set [hasMenu] to `true` and override the [setupMenu] method.
+ *
+ * @param B Type of data binding class corresponding [layoutId]
+ * @property toolbarId The Toolbar ID
+ * @property hasBackNavigation Indicates that the toolbar contains back navigation. With this option
+ * there is no need to add back navigation in layout and it will be added automatically with default
+ * layout and content description. Default value is set to false.
+ * @property menuId The menu resource ID
+ * @property hasMenu Indicates that the toolbar contains Options menu
+ * @param layoutId The layout resource ID of the layout to inflate
+ * @author eMan a.s.
+ * @see 0.9.0
+ */
+@Suppress("MemberVisibilityCanBePrivate")
+open class KaalToolbarBindingFragment<B : ViewDataBinding>(
+    @IdRes protected val toolbarId: Int? = null,
+    protected val hasBackNavigation: Boolean = false,
+    @MenuRes protected val menuId: Int? = null,
+    protected val hasMenu: Boolean = menuId != null,
+    @LayoutRes layoutId: Int
+) : KaalBindingFragment<B>(layoutId) {
+
+    private var wrappedToolbar: ActionBar? = null
+
+    /**
+     * Provides reference to Toolbar inside fragment's View. If this method is not overridden, a [toolbarId] must be
+     * provided in the constructor.
+     *
+     * > **Note:** Overridden method should not call `super()`.
+     */
+    open fun findToolbar(): Toolbar = requireNotNull(requireView().findViewById(
```

Comment:
It was only for backward compatibility in the ČSOB project. I would remove it or set it as private and final.